src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -0,0 +1,1154 @@
+                    varDefAst.findFirstToken(TokenTypes.TYPE), findScopeOfVariable(varDefAst));
+            classAstToClassDesc.get(parentAst.getParent()).addInstOrClassVar(desc);
+        }
+    }
+
+    /**
+     * Whether instance variable or class variable has private access modifier.
+     *
+     * @param varDefAst ast node of type {@link TokenTypes#VARIABLE_DEF}
+     * @return true if instance variable or class variable has private access modifier
+     */
+    private static boolean isPrivateInstanceVariable(DetailAST varDefAst) {
+        return CheckUtil.getAccessModifierFromModifiersToken(
+                varDefAst) == AccessModifierOption.PRIVATE;
+    }
+
+    /**
+     * Get the {@link ClassDesc} of the super class of anonymous inner class.
+     *
+     * @param literalNewAst ast node of type {@link TokenTypes#LITERAL_NEW}
+     * @return {@link ClassDesc} of the super class of anonymous inner class
+     */
+    private ClassDesc getSuperClassOfAnonInnerClass(DetailAST literalNewAst) {
+        ClassDesc obtainedClass = null;
+        final String shortNameOfClass = getShortNameOfAnonInnerClass(literalNewAst);
+
+        final List<ClassDesc> classesWithSameName = classesWithSameName(shortNameOfClass);
+        if (!classesWithSameName.isEmpty()) {
+            obtainedClass = getTheNearestClass(
+                    anonInnerAstToClassDesc.get(literalNewAst).getQualifiedName(),
+                    classesWithSameName);
+        }
+        return obtainedClass;
+    }
+
+    /**
+     * Get the short name of super class of anonymous inner class.
+     * Example-
+     * <pre>
+     * TestClass.NestedClass obj = new Test().new NestedClass() {};
+     * // Short name will be Test.NestedClass
+     * </pre>
+     *
+     * @param literalNewAst ast node of type {@link TokenTypes#LITERAL_NEW}
+     * @return short name of base class of anonymous inner class
+     */
+    public static String getShortNameOfAnonInnerClass(DetailAST literalNewAst) {
+        DetailAST parentAst = literalNewAst.getParent();
+        while (TokenUtil.isOfType(parentAst, TokenTypes.LITERAL_NEW, TokenTypes.DOT)) {
+            parentAst = parentAst.getParent();
+        }
+        final DetailAST firstChild = parentAst.getFirstChild();
+        return extractQualifiedName(firstChild);
+    }
+
+    /**
+     * Add non-private instance variables of the super class of the anonymous class
+     * to the variables stack.
+     *
+     * @param obtainedClass super class of the anon inner class
+     * @param variablesStack stack of all the relevant variables in the scope
+     * @param literalNewAst ast node of type {@link TokenTypes#LITERAL_NEW}
+     */
+    private void modifyVariablesStack(ClassDesc obtainedClass,
+            Deque<VariableDesc> variablesStack,
+            DetailAST literalNewAst) {
+        if (obtainedClass != null) {
+            final Deque<VariableDesc> instAndClassVarDeque = classAstToClassDesc.get(
```

Comment:
Can we do:

```
            final Deque<VariableDesc> instAndClassVarDeque = classesWithInstAndClassVar
                              .get(obtainedClass.getClassOrInterfaceAst())
                              .getUpdatedCopyOfVarStack(literalNewAst);


```

I think this is easier to read.

Reply:
Yes, it is definitely better. Done.

Reply:
Resolved.